liquibase-core/src/main/java/liquibase/command/core/TagCommandStep.java

```diff
@@ -1,52 +1,54 @@
-import liquibase.configuration.ConfigurationValueObfuscator;
-import liquibase.exception.CommandExecutionException;
+import liquibase.database.Database;
+import liquibase.exception.LockException;
+import liquibase.lockservice.LockService;
+import liquibase.lockservice.LockServiceFactory;
 
-public class TagCommandStep extends AbstractCliWrapperCommandStep {
+public class TagCommandStep extends AbstractCommandStep {
 
     public static final String[] COMMAND_NAME = {"tag"};
 
-    public static final CommandArgumentDefinition<String> URL_ARG;
-    public static final CommandArgumentDefinition<String> DEFAULT_SCHEMA_NAME_ARG;
-    public static final CommandArgumentDefinition<String> DEFAULT_CATALOG_NAME_ARG;
-    public static final CommandArgumentDefinition<String> USERNAME_ARG;
-    public static final CommandArgumentDefinition<String> PASSWORD_ARG;
     public static final CommandArgumentDefinition<String> TAG_ARG;
-    public static final CommandArgumentDefinition<String> DRIVER_ARG;
-    public static final CommandArgumentDefinition<String> DRIVER_PROPERTIES_FILE_ARG;
+    public static final CommandArgumentDefinition<Database> DATABASE_ARG;
 
     static {
         CommandBuilder builder = new CommandBuilder(COMMAND_NAME);
-        URL_ARG = builder.argument(CommonArgumentNames.URL, String.class).required()
-                .description("The JDBC database connection URL").build();
-        DEFAULT_SCHEMA_NAME_ARG = builder.argument("defaultSchemaName", String.class)
-                .description("The default schema name to use for the database connection").build();
-        DEFAULT_CATALOG_NAME_ARG = builder.argument("defaultCatalogName", String.class)
-                .description("The default catalog name to use for the database connection").build();
-        DRIVER_ARG = builder.argument("driver", String.class)
-                .description("The JDBC driver class").build();
-        DRIVER_PROPERTIES_FILE_ARG = builder.argument("driverPropertiesFile", String.class)
-               .description("The JDBC driver properties file").build();
-        USERNAME_ARG = builder.argument(CommonArgumentNames.USERNAME, String.class)
-                .description("Username to use to connect to the database").build();
-        PASSWORD_ARG = builder.argument(CommonArgumentNames.PASSWORD, String.class)
-                .description("Password to use to connect to the database")
-                .setValueObfuscator(ConfigurationValueObfuscator.STANDARD)
-                .build();
-        TAG_ARG = builder.argument("tag", String.class).required()
-                .description("Tag to add to the database changelog table").build();
+        TAG_ARG = builder.argument("tag", String.class).required().description("Tag to add to the database changelog table").build();
+        DATABASE_ARG = builder.argument("database", Database.class).hidden().description("Database connection").build();
+        InternalDatabaseCommandStep.addApplicableCommand(COMMAND_NAME);
```

Comment:
We discussed this during our zoom meeting, copying information here for continuity.

It would be preferred to add another method to the `CommandBuilder` class that abstracts away both of these lines. It could look something like this:

`DATABASE_ARG = builder.databaseArgument().build()`